gammapy/scripts/analysis.py

```diff
@@ -51,6 +63,118 @@ def settings(self):
+        elif datastore_path.is_dir():
+            datastore = DataStore().from_dir(datastore_path)
+        else:
+            raise FileNotFoundError("Datastore {} not found.".format(datastore_path))
+
+        ids = set()
+        selection = dict()
+        for criteria in self.settings["observations"]["filters"]:
+            selected_obs = ObservationTable()
+
+            # -- TODO Handled by datastore.obs_table.select_observations
+            # -
+            selection["type"] = criteria["filter_type"]
+            for key, val in criteria.items():
+                if key in ["lon", "lat", "radius", "border"]:
+                    val = Angle(val)
+                selection[key] = val
+            if selection["type"] == "angle_box":
+                selection["type"] = "par_box"
+                selection["value_range"] = Angle(criteria["value_range"])
+            if selection["type"] == "sky_circle" or selection["type"].endswith("_box"):
+                selected_obs = datastore.obs_table.select_observations(selection)
+            if selection["type"] == "par_value":
+                mask = (
+                    datastore.obs_table[criteria["variable"]] == criteria["par_value"]
+                )
+                selected_obs = datastore.obs_table[mask]
+            if selection["type"] == "ids":
+                obs_list = datastore.get_observations(criteria["obs_ids"])
+                selected_obs["OBS_ID"] = [obs.obs_id for obs in obs_list.list]
+            if selection["type"] == "all":
+                obs_list = datastore.get_observations()
+                selected_obs["OBS_ID"] = [obs.obs_id for obs in obs_list.list]
+            # --
+            # -
+
+            if len(selected_obs):
+                if "exclude" in criteria and criteria["exclude"]:
+                    ids.difference_update(selected_obs["OBS_ID"].tolist())
+                else:
+                    ids.update(selected_obs["OBS_ID"].tolist())
+        if len(ids):
+            self.observations = datastore.get_observations(ids)
+        else:
+            self.observations = Observations()
+
+    def reduce_data(self):
+        """Produce reduced data sets."""
+        self.config.validate()
+
+        # create geometry
+        self.geom = WcsGeom.create(
+            skydir=tuple(self.settings["geometry"]["skydir"]),
+            binsz=self.settings["geometry"]["binsz"],
+            width=tuple(self.settings["geometry"]["width"]),
+            coordsys=self.settings["geometry"]["coordsys"],
+            proj=self.settings["geometry"]["proj"],
+        )
+        # axes=[energy_axis]
+
+        if self.settings["reduction"]["data_reducer"] == "1d":
+            self._spectrum_extraction()
+
+    def _spectrum_extraction(self):
+        # """Run all steps for the spectrum extraction."""
+
+        on = self.settings["reduction"]["background"]["on_region"]
+        on_lon = Angle(on["center"][0])
```

Comment:
Is it possible to move the handling of Quantity, Angle, SkyCoord, Time, SkyRegion into the config file handling in a generic way? I.e. declare those as types in the schema, and then the right object creation is done automatically, either by yaml or jsonschema somewhere?
I think if you use this YAML load (http://docs.astropy.org/en/stable/io/misc.html#example) it will do the trick.
But I don't know how this interplays with json-schema.
What we want is to be able to write strings like "30 deg" or "42 rad" for angles in the config file, and not need to put the `Angle` to type-cast in the Analysis code and elsewehre in Gammapy.
Possible?

Reply:
As it was discussed in last dev call, this will be done in another PR.